evm/src/main/java/org/hyperledger/besu/evm/precompile/AltBN128MulPrecompiledContract.java

```diff
@@ -148,4 +162,8 @@ private static BigInteger extractParameter(
     final byte[] raw = Arrays.copyOfRange(input.toArrayUnsafe(), offset, offset + length);
     return new BigInteger(1, raw);
   }
+
+  private static Integer getCacheKey(final Bytes input) {
```

Comment:
I suggest to define it only once in `AbstractPrecompiledContract.java` and reference it in all the cached precompiles.

Reply:
Seems reasonable - I will define it there and override it for ECRECOVER.  I think that is the only one that is different

Reply:
actually I think ECREC can use the same cache function